judge/views/problem.py

```diff
@@ -536,8 +536,11 @@ def problem_submit(request, problem, submission=None):
             return HttpResponseForbidden('<h1>Do you want me to ban you?</h1>')
         raise Http404()
 
+    can_edit_problem = problem.is_editable_by(request.user)
+
     if request.method == 'POST':
-        form = ProblemSubmitForm(request.POST, instance=Submission(user=profile, problem=problem))
+        form = ProblemSubmitForm(request.POST, can_edit_problem=can_edit_problem,
```

Comment:
Is it possible to fake `can_edit_problem`, given that it is passed to the form without validation?

Reply:
Should not be possible, since `can_edit_problem` is taken directly from `problem.is_editable_by`.